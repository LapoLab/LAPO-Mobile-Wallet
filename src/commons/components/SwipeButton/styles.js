// @flow
import { Dimensions } from 'react-native';

const widthWindow = Dimensions.get('window').width;

export default {
  buttonContainer: {
    backgroundColor: '#BBCDEB',
    borderRadius: 30,
    height: 64,
    width: '80%',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start',
    paddingHorizontal: 10,
  },
  button: {
    width: 52,
    height: 52,
    borderRadius: 26,
    backgroundColor: '#4183BD',
    alignItems: 'center',
    justifyContent: 'center',
  },
  SendImage: {
    width: 24,
    resizeMode: 'contain',
  },
  backgroundActive: {
    width: '98%',
    height: 52,
    borderRadius: 26,
    left: 8,
    backgroundColor: '#4183BD',
    position: 'absolute',
  },
  infoContainer: {
    height: 56,
    width: widthWindow * 0.80,
    top: 22,
    position: 'absolute',
    alignItems: 'center',
  },
  textSlideToSend: {
    fontFamily: 'Lato-Regular',
    fontSize: 18,
    color: 'white',
  },
  imageStrelka: {
    width: 60,
    top: -4,
    resizeMode: 'contain',
  },
};
