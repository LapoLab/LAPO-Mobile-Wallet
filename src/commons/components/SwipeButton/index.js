/**
 * @flow
 */

import React, { Component } from 'react';
import {
  View,
  PanResponder,
  Dimensions,
  Image,
  Text,
} from 'react-native';
import send from '../../../assets/sendButton.png';
import styles from './styles';

const { width: widthWindow } = Dimensions.get('window');

type Props = {
  onUnlock: Function,
};
type State = {
  pos: number,
};
class SwipeButton extends Component<Props, State> {
  _panResponder: Function;

  constructor(props: Props) {
    super(props);
    const { onUnlock } = props;
    this._panResponder = PanResponder.create({
      // Ask to be the responder:
      onStartShouldSetPanResponder: () => true,
      onStartShouldSetPanResponderCapture: () => true,
      onMoveShouldSetPanResponder: () => true,
      onMoveShouldSetPanResponderCapture: () => true,

      onPanResponderMove: (evt, gestureState) => {
        const pos = Math.abs((gestureState.dx * 100) / widthWindow) + 10;
        if (pos <= 84) {
          this.setState({ pos });
        }
        if (pos > 84 && !this.isActive) {
          onUnlock();
          this.setState({ pos: 0 });
          this.isActive = true;
        }
      },
      onPanResponderTerminationRequest: () => true,
      onPanResponderRelease: () => {
        this.setState({ pos: 0 });
        this.isActive = false;
      },
      onShouldBlockNativeResponder: () => true,
    });
  }

  state = {
    pos: 0,
  };

  isActive = false;

  render() {
    const { pos } = this.state;
    return (
      <View style={styles.buttonContainer}>
        <View style={styles.infoContainer}>
          <Text style={styles.textSlideToSend}>SLIDE TO SEND</Text>
        </View>
        {pos > 5 && (
          <View
            style={{
              ...styles.backgroundActive,
              width: `${pos + 17}%`,
            }}
          />
        )}
        <View
          style={{
            ...styles.button,
            marginLeft: `${pos - 1}%`,
          }}
          {...this._panResponder.panHandlers}
        >
          <Image style={styles.SendImage} source={send} />
        </View>
      </View>
    );
  }
}

export default SwipeButton;
