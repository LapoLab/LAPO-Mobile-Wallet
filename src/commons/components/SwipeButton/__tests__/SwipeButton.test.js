import React from 'react';
import renderer from 'react-test-renderer';
import SwipeButton from '../index';

describe('SwipeButton component', () => {
  test('renders snapshots default', () => {
    const tree = renderer.create(<SwipeButton />);
    const render = tree.toJSON();
    expect(render).toMatchSnapshot();
  });
});
