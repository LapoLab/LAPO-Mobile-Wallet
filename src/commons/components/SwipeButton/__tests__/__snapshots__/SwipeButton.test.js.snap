// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwipeButton component renders snapshots default 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#BBCDEB",
      "borderRadius": 30,
      "flexDirection": "row",
      "height": 64,
      "justifyContent": "flex-start",
      "paddingHorizontal": 10,
      "width": "80%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 56,
        "position": "absolute",
        "top": 22,
        "width": 600,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "Lato-Regular",
          "fontSize": 18,
        }
      }
    >
      SLIDE TO SEND
    </Text>
  </View>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#4183BD",
        "borderRadius": 26,
        "height": 52,
        "justifyContent": "center",
        "marginLeft": "-1%",
        "width": 52,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/sendButton.png",
        }
      }
      style={
        Object {
          "resizeMode": "contain",
          "width": 24,
        }
      }
    />
  </View>
</View>
`;
